awx/ui_next/src/components/LaunchPrompt/SurveyStep.jsx

```diff
@@ -1,7 +1,204 @@
+    question.max ? maxLength(question.max, i18n) : null,
+  ];
+  return (
+    <FormField
+      id={`survey-question-${question.variable}`}
+      name={question.variable}
+      label={question.question_name}
+      tooltip={question.question_description}
+      isRequired={question.required}
+      validate={combine(validators)}
+      type={question.type}
+      minLength={question.min}
+      maxLength={question.max}
+    />
+  );
+}
+
+function NumberField({ question, i18n }) {
+  const validators = [
+    question.required ? required(null, i18n) : null,
+    minMaxValue(question.min, question.max, i18n),
+    question.type === 'integer' ? integer(i18n) : null,
+  ];
+  return (
+    <FormField
+      id={`survey-question-${question.variable}`}
+      name={question.variable}
+      label={question.question_name}
+      tooltip={question.question_description}
+      isRequired={question.required}
+      validate={combine(validators)}
+      type="number"
+      min={question.min}
+      max={question.max}
+    />
+  );
+}
+
+function MultipleChoiceField({ question }) {
+  const [field, meta] = useField(question.variable);
+  const id = `survey-question-${question.variable}`;
+  const isValid = !(meta.touched && meta.error);
+  return (
+    <FormGroup
+      fieldId={id}
+      helperTextInvalid={meta.error}
+      isRequired={question.required}
+      isValid={isValid}
+      label={question.question_name}
+    >
+      <FieldTooltip content={question.question_description} />
+      <AnsibleSelect
+        id={id}
+        isValid={isValid}
+        {...field}
+        data={question.choices.split('\n').map(opt => ({
+          key: opt,
+          value: opt,
+          label: opt,
+        }))}
+      />
+    </FormGroup>
+  );
+}
+
+function MultiSelectField({ question }) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [field, meta, helpers] = useField(question.variable);
+  const id = `survey-question-${question.variable}`;
+  const isValid = !(meta.touched && meta.error);
+  return (
+    <FormGroup
+      fieldId={id}
+      helperTextInvalid={meta.error}
+      isRequired={question.required}
+      isValid={isValid}
+      label={question.question_name}
+    >
+      <FieldTooltip content={question.question_description} />
+      <Select
+        variant={SelectVariant.typeaheadMulti}
+        id={id}
+        onToggle={setIsOpen}
+        onSelect={(event, option) => {
+          if (field.value.includes(option)) {
+            helpers.setValue(field.value.filter(o => o !== option));
+          } else {
+            helpers.setValue(field.value.concat(option));
+          }
+        }}
+        isExpanded={isOpen}
+        selections={field.value}
+      >
+        {question.choices.split('\n').map(opt => (
+          <SelectOption key={opt} value={opt} />
+        ))}
+      </Select>
+    </FormGroup>
+  );
 }
 
-export default InventoryStep;
+export default withI18n()(SurveyStep);
```

Comment:
Nice...it'd be cool if we could share some of this when we get to the credential form "generator" (though I don't think they are the same api spec so that might not be feasible.

Reply:
I really like how tight and simple this is.